bats/tests/profile/sample.bats

```diff
@@ -25,69 +43,218 @@ start_app() {
     wait_for_container_engine
 }
 
-verify_settings() {
-    PROFILE_TYPE=$PROFILE_LOCKED
-    run profile_exists
-    "${assert}_success"
-
-    PROFILE_TYPE=$PROFILE_DEFAULTS
-    run profile_exists
-    "${assert}_success"
+verify_profiles() {
+    local PROFILE_TYPE
+    for PROFILE_TYPE in "$PROFILE_LOCKED" "$PROFILE_DEFAULTS"; do
+        run profile_exists
+        "${assert}_success"
+    done
+}
 
+verify_settings() {
+    # settings from defaults profile
     run get_setting .containerEngine.name
-    "${assert}_output" "$PROFILE_CONTAINER_ENGINE"
+    "${assert}_output" "$DEFAULTS_CONTAINER_ENGINE_NAME"
 
     run get_setting .application.startInBackground
-    "${assert}_output" "$PROFILE_START_IN_BACKGROUND"
+    "${assert}_output" "$DEFAULTS_START_IN_BACKGROUND"
+
+    # settings from locked profile
+    run get_setting .containerEngine.allowedImages.enabled
+    "${assert}_output" "$LOCKED_ALLOWED_IMAGES_ENABLED"
+
+    run get_setting .containerEngine.allowedImages.patterns
+    "${assert}_output" --partial "${LOCKED_ALLOWED_IMAGES_PATTERNS[@]}"
+
+    run get_setting .application.extensions.allowed.enabled
+    "${assert}_output" "$LOCKED_EXTENSIONS_ALLOWED_ENABLED"
+
+    run get_setting .application.extensions.allowed.list
+    "${assert}_output" --partial "${LOCKED_EXTENSIONS_ALLOWED_LIST[@]}"
+
+    run get_setting .kubernetes.version
+    "${assert}_output" "$LOCKED_KUBERNETES_VERSION"
+    refute_output "$DEFAULTS_KUBERNETES_VERSION"
+}
+
+install_extensions() {
+    run rdctl extension install "$FORBIDDEN_EXTENSION"
+    "${refute}_success"
+
+    run rdctl extension install "$ALLOWED_EXTENSION_NAME:$FORBIDDEN_EXTENSION_TAG"
+    "${refute}_success"
+
+    run rdctl extension install "${LOCKED_EXTENSIONS_ALLOWED_LIST[0]}"
+    assert_success
 }
 
 @test 'initial factory reset' {
     factory_reset
 }
 
-@test 'start up without profile' {
+@test 'start up with NO profiles' {
     RD_USE_PROFILE=false
-    start_app
+    RD_USE_IMAGE_ALLOW_LIST=false
+    start_application
 }
 
-@test 'verify default settings' {
+@test 'verify there were NO profiles created' {
+    before verify_profiles
+}
+
+@test 'verify default settings were applied' {
     before verify_settings
 }
 
-@test 'factory reset' {
+@test 'verify all extensions can be installed' {
+    #WORKAROUND `rdctl` tries to install extensions before app is ready.
+    wait_for_container_engine
+    sleep 30
+    before install_extensions
+}
+
+@test 'factory reset before creating profiles' {
     factory_reset
 }
 
-@test 'create profile' {
+@test 'create profiles' {
     PROFILE_TYPE=$PROFILE_LOCKED
     create_profile
```

Comment:
No, this won't work. `PROFILE_TYPE` determines for **all** profile functions which profile they work on, so it needs to be set for all of them. It is like a "docker context" but for profiles...

Reply:
We set `PROFILE_TYPE=$PROFILE_DEFAULTS` immediately after; presumably this doesn't last past that line?

I'd rather we set it at each caller so that it's clear what is meant to be in effect at each point…

(But yes, the bigger issue is that we're passing parameters through environment variables, which makes things confusing at the site where it's used.)

Reply:
As I wrote above, I view setting `PROFILE_TYPE` (and `PROFILE_LOCATION`) similar to calling `kubectl config use-context $CONTEXT`.

Yes, you can use `kubectl --context $CONTEXT ...` to pass it explicitly, but I still prefer setting it globally once, just to use it consistently.